.github/workflows/rust-clippy.yml

```
@@ -0,0 +1,38 @@
+name:                         Rust clippy
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+      - stable
+
+jobs:
+  linter:
+    name:                     Cargo clippy
+    runs-on:                  ubuntu-latest
+    steps:
+
+      - name:                 Cancel Previous Runs
+        uses:                 styfle/cancel-workflow-action@0.12.1
+        with:
+          access_token:       ${{ github.token }}
+
+      - name:                 Checkout sources
+        uses:                 actions/checkout@v4.1.1
+        with:
+          fetch-depth:        50
+          submodules:         'recursive'
+
+      - name:                 Install Rust stable toolchain
+        uses:                 actions-rs/toolchain@v1.0.7
+        with:
+          profile:            minimal
+          toolchain:          stable
+          override:           true
+
+      - name:                 Rust Cache
+        uses:                 Swatinem/rust-cache@v2.7.3
+
+      - name:                 cargo clippy
+        run:                  cargo clippy -- -D warnings -D clippy::pedantic -D clippy::correctness -D clippy::complexity -D clippy::perf
```

important thing we forgot!
```suggestion
        run:                  cargo clippy --all-targets -- -D warnings -D clippy::pedantic -D clippy::correctness -D clippy::complexity -D clippy::perf
```